### PR TITLE
Improved --help and added --version

### PR DESCRIPTION
**Pull request by N1xis10t**
I implemented a --version longoption that prints version information, as recommended by the GNU Coding Standards.

I also made changes to the helpfile:
~ It now fits in small terminals that are 80 chars wide without being screen wrapped
~ It looks pretty like GNU utilities now
~ It has a couple less typos
~ It now has a link to the mkp224o homepage and a link to the issues page
~ And it now mentions the --help and --usage longoptions (and also the new --version longoption that I added).

### CODE REVIEW

**Comment by cathugger** [comment]
ohi, sorry for long delay.

I don't particularly care following GNU Coding Standards, but having `--version` sounds kinda reasonable.

`-f, =FILTERFILE` - is there documented convention somewhere for this? it looks kinda easier to mechanically parse but I'm not sure if I like the change. I've not seen that used anywhere so it may actually be more confusing if it's different from what people are used to.
I see you also have replaced tabs with 2 spaces, which is probably fine i guess, though I liked neatness of using only single character there.
I also see that you added `{}` around rawyaml condition. please don't do that, such rule should either be applied uniformly or not applied at all (and I prefer second case obviously since I wrote it that way).
I also noticed that you've mixed tabs and spaces for indention in some lines of source code. please fix that -- `.editorconfig` provided should already configure editors which support that to use tabs, but some editors don't support that or not by default, in which case manual config may be needed.

above issues aside, I'm not sure whether I like current way of specifying version string.
first of all, it's probably not "1.5.0" anymore, there's been changes since. some quite good changes. maybe "1.5.0+git"?
or have a way to find out git commit hash and display that?
having something like "1.6.0-dev" would be inaccurate as I don't really know beforehand whether it'll be "1.5.1" or "1.6.0".
release tarballs don't have git history, so probably need some way to specify it there too.
kinda sane way to do this would be detecting this in configure script:
1. if there is version.txt, use content of that
2. if there is .git folder, use git commit hash (revision number and latest tag combination could also be helpful, there's more than one way to do this)
3. otherwise use "unknown"

you can try implementing it that way or just fix other smaller things and let me do that.

**Comment by N1xis10t** [comment]
To address the two style/formatting things:

- `-f, =FILTERFILE`: I did not do this based on a documented convention, and I haven't seen other programs do this. What I _have_ seen is stuff like this: `-r, --prompt=STRING` (this was in `man --help`). Having the little `=` sign there even though there isn't a longoption seemed sensible to me. How would you prefer it be formatted? `-f, FILTERFILE`?
- ¯\\_ (ツ)_/¯ My terminal was rendering tabs significantly wider than two spaces, and I thought that two spaces looked nicer.

**Comment by N1xis10t** [comment]
As for the version string, I totally agree with you. Your idea seems like a great way to implement it. I might end up hacking something together at some point, but I don't have much experience with shell scripts so I feel more comfortable letting you do that.

**Comment by cathugger** [comment]
>-f, =FILTERFILE: I did not do this based on a documented convention, and I haven't seen other programs do this. What I have seen is stuff like this: -r, --prompt=STRING (this was in man --help). Having the little = sign there even though there isn't a longoption seemed sensible to me.

that's because they allow literally `--option=value` while my handwritten parsing can only do `--option value`. `=` makes no sense here.

>How would you prefer it be formatted? -f, FILTERFILE?

that `,` makes no sense. dunno, maybe `-f FILTERFILE`? in general, everywhere GNU stuff has argument which can take free-style value, they provide longopt. that isn't true in mkp224o's case.

>¯\\_ (ツ)_/¯ My terminal was rendering tabs significantly wider than two spaces, and I thought that two spaces looked nicer.

I think I had my editor configured to use 3 spaces for tab back when I initially wrote this. these are very often configurable which is why I like using them.

>I might end up hacking something together at some point, but I don't have much experience with shell scripts so I feel more comfortable letting you do that.

alright. will probably do that over the weekend or something.

**Comment by cathugger** [comment]
oh right, also we probably want `-V` be the same as `--version`.
and `printversion` shouldn't take `FILE *` as the only thing passed to it can be `stdout`.
`printhelp` uses `stdout` and `stderr` depending on some stuff so it needs to be configurable there, while that's not the case for `printversion`.

**Comment by N1xis10t** [comment]
Did it

File: main.c [hunk 1]
```diff
@@ -122,7 +122,7 @@ static void printhelp(FILE *out,const char *progname)
 		"\t                      (may be useful for tor controller API)\n"
 		"\t-h --help           print help to stdout and quit\n"
 		"\t   --usage          same as --help\n"
-		"\t   --version        print version information to stdout and exit\n"
+		"\t-V --version        print version information to stdout and exit\n"
```
Comment (cathugger) [inline main.c]: `-h, --help`, `-V, --version`

**Comment by cathugger** [comment]
tbh 2 spaces instead of `\t` was kinda fine. probably even better for preserving screen space.
I also don't like 2 spaces additional shift for continuation lines, various `--help`s of things I've tested didn't have that and it looks weird.
otherwise, content of `printversion` is sorta weird regarding attribution, should probably rewrite some of it but I'll probably do that myself.
i think it'll be fine now otherwise and I will figure rest out myself.

**Comment by cathugger** [comment]
current stuff is fine for me to work on, now i just gotta find some free time when im not feeling too dead...

**Comment by N1xis10t** [comment]
For sure

**Comment by N1xis10t** [comment]
Don't run yourself into the ground

**Comment by cathugger** [comment]
i think i have local branch with this mostly doing stuff properly plus a lot of other unfinished changes.
ill get to this again at some point probably, my head's just been totally elsewhere recently.

**Comment by JAKAMI99** [comment]
> i think i have local branch with this mostly doing stuff properly plus a lot of other unfinished changes. ill get to this again at some point probably, my head's just been totally elsewhere recently.

Make sure to watch yourself, take your time <3
Thanks for this awesome project :) helped me alot

**Comment by cathugger** [comment]
I've committed this minus homepage mentions in master now so i guess i can close.

**Pull request closed without merging**